### DISCUSSION TITLE
how to change map to clickhouse Nested structure?

### DISCUSSION THREAD

**Discussion by dengc367** [vectordotdev/vector#11834] [Q&A] [answered]
I use vector to `parse_json` from json file to clickhouse,  because the json  like below

`{"res":"5e60bbc17f068d0ba4f156e5","_id":"622e150183d43d02dca5139b","properties":{"key1":"value1", "key2":2, "key3": 3.0}}`

and the clickhouse columns have a `Nested` structure like below:
```
res String,
_id String,
properties.key Array(Nullable(String)),
properties.value Array(Nullable(String)),
```

so I need change the above json to the clickhouse format:

`{"res":"5e60bbc17f068d0ba4f156e5","_id":"622e150183d43d02dca5139b","properties.key": ["key1","key2","key3"],"properties.value": ["value1","2","3.0"]`

@binarylogic  how to transform it use the `vector vrl`?

**Comment by jszwedko** [comment, marked as answer]
Answered over here: https://github.com/vectordotdev/vector/issues/11848#issuecomment-1071621047
